src/components/VideoProvider/useLocalTracks/useLocalTracks.ts

```diff
@@ -1,13 +1,31 @@
 import { useCallback, useEffect, useState } from 'react';
 import Video, { LocalVideoTrack, LocalAudioTrack, CreateLocalTrackOptions } from 'twilio-video';
 
+// This function ensures that the user has granted the browser permission to use audio and video
+// devices. If permission has not been granted, it will cause the browser to ask for permission
+// for audio and video at the same time (as opposed to separate requests).
+function ensureMediaPermissions() {
```

Comment:
How do we handle the error for the following cases?
- user denies permission
- user doesn't have camera but have mic
- user doesn't have mic but have camera

Reply:
Great question! This PR doesn't change how the app behaves when the user denies permissions. When permissions are denied, there is an uncaught `Permission Denied` error, and then the app behaves as if there are no audio or video devices available.

Obviously this isn't ideal. We have a ticket to implement a good UI for handling denied permissions: https://issues.corp.twilio.com/browse/AHOYAPPS-564. I think this is a ticket to prioritize, but I would like to complete it in a separate PR.